src/main/java/com/boot/service/ItemService.java

```diff
@@ -98,35 +94,30 @@
     }
 
 
-    public ResponseEntity<?> buyItem() {
-//        UserDetailsImpl user = (UserDetailsImpl) org.springframework.security.core.context.SecurityContextHolder
-//                .getContext().getAuthentication().getPrincipal();
+    public ResponseEntity<?> buyItem(User user) {
 
-        //get information about authenticated user
-        Authentication loggedInUser = SecurityContextHolder.getContext().getAuthentication();
-        User loggedInUserFromDB = userRepository.findAllByUsername(loggedInUser.getName());
-        String userEmail = loggedInUserFromDB.getEmail();
+        String userEmail = user.getEmail();
 
         //check that the users basket not empty
-        List<Basket> userItemList = basketRepository.findItemByUserId(loggedInUserFromDB.getId());
+        List<Basket> userItemList = basketRepository.findItemByUserId(user.getId());
         if (userItemList.size() == 0) {
             return ResponseEntity.badRequest().body(new MessageResponse("The basket is empty!"));
         }
 
         //get full information about item
-        List<String> namesOfUsersItems = userItemList.stream().map(item -> item.getItem()).collect(Collectors.toList());
+        List<Item> namesOfUsersItems = userItemList.stream().map(item -> item.getItem()).collect(Collectors.toList());
         List<String> fullInformationAboutItem = new ArrayList<>();
```

Comment:
fullInformationAboutItem - is not full information about smth. It is just string array.  According to code below, i understood that the list contains item.name and item.description. I have to keep in mind that each 1,3,5... element in the array - name and each 2,4,6 - description.  Instead you could wrap the information to object and keep List<ItemInfo>  or List<ItemShort> . Each such object will have methods : getName(), getDescription()